packages/foam-vscode/src/features/commands/convert-links-format-in-note.ts

```diff
@@ -0,0 +1,179 @@
+import { commands, ExtensionContext, window } from 'vscode';
+import { isMdEditor } from '../../utils';
+import { Foam } from '../../core/model/foam';
+import { FoamWorkspace } from '../../core/model/workspace';
+import { fromVsCodeUri, toVsCodeRange } from '../../utils/vsc-utils';
+import { ResourceParser } from '../../core/model/note';
+import { IMatcher } from '../../core/services/datastore';
+import { convertLinkFormat } from '../../core/janitor';
+const vscode = require('vscode'); /* cannot import workspace from above statement and not sure what happened */
+
+enum ConvertOption {
+  Wikilink2MDlink,
+  MDlink2Wikilink,
+}
+
+interface IConfig {
+  from: string;
+  to: string;
+}
+
+const Config: { [key in ConvertOption]: IConfig } = {
+  [ConvertOption.Wikilink2MDlink]: {
+    from: 'wikilink',
+    to: 'link',
+  },
+  [ConvertOption.MDlink2Wikilink]: {
+    from: 'link',
+    to: 'wikilink',
+  },
+};
+
+export default async function activate(
+  context: ExtensionContext,
+  foamPromise: Promise<Foam>
+) {
+  const foam = await foamPromise;
+
+  /* 
+  commands:
+  foam-vscode.convert-wikilink-to-markdownlink-inplace
+  foam-vscode.convert-markdownlink-to-wikilink-inplace
+  foam-vscode.convert-wikilink-to-markdownlink-in-copy
+  foam-vscode.convert-markdownlink-to-wikilink-in-copy
+   */
```

Comment:
I'd suggest having only one command, something like `foam-vscode.convert-link-style`, with a `linkStyle` property (which defaults to `markdown-link`). Running the command on a file with the links already in that style would result in no change, otherwise the changes are applied.

Reply:
done